### PR TITLE
Chore/584 testing library jest setup

### LINKED ISSUE #584: Add jest testing-library setup and one basic test
-  Add https://testing-library.com and related packages to the project
-  Create one simple test
-  Make setup work

To enable people to easily write unit/intergration tests.  
As discussed in the [slack channel](https://bitcoindesign.slack.com/archives/C02591ADXM2/p1644046753124729).

### PR DESCRIPTION
**Pull request by escapedcat**
Fixes #584 

- [x] Add https://testing-library.com/ and related packages to the project
- [x] Create two simple tests
   - [x] `BalanceCard`
   - [x] `Button`
- [x] Make setup work
- [x] Create test for `ConfirmPayment`
  - [x] react-router-dom
  - [x] ExpectedLnPrefix
  - [x] Expected signature to be an Uint8Array
  - [x] react-palette

#### react-router-dom ✅

```sh
  FAIL  src/app/screens/ConfirmPayment/index.test.tsx (9.21 s)

  ConfirmPayment
    ✕ render (131 ms)

  ● ConfirmPayment › render

    useLocation() may be used only in the context of a <Router> component.
```

#### ExpectedLnPrefix ✅
    
```sh
      30 |   const auth = useAuth();
      31 |   const invoiceRef = useRef(
    > 32 |     parsePaymentRequest({
         |     ^
      33 |       request:
      34 |         props.paymentRequest || (searchParams.get("paymentRequest") as string),
      35 |     })
```

See: https://github.com/alexbosworth/invoices/blob/master/bolt11/parse_payment_request.js#L68

#### Expected signature to be an Uint8Array ✅

```sh
  ● ConfirmPayment › render

    Expected signature to be an Uint8Array

      31 |   const auth = useAuth();
      32 |   const invoiceRef = useRef(
    > 33 |     parsePaymentRequest({
         |     ^
      34 |       request:
      35 |         props.paymentRequest || (searchParams.get("paymentRequest") as string),
      36 |     })
```

#### react-palette ✅
[`react-palette`](https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/components/PublisherCard/index.tsx#L1) is causing this error, so commented it out for now, till everyhting else works:

```sh
  ● Test suite failed to run

    TypeError: (0 , _png.default) is not a function

    > 1 | import { usePalette } from "react-palette";
        | ^
      2 | import { LinkIcon } from "@bitcoin-design/bitcoin-icons-react/filled";
      3 |
      4 | const DEFAULT_IMAGE =

      at typeModule (node_modules/@jimp/types/src/index.js:9:40)
      at forEach (node_modules/@jimp/custom/src/index.js:27:18)
          at Array.forEach (<anonymous>)
      at Object.configure [as default] (node_modules/@jimp/custom/src/index.js:50:25)
      at Object.<anonymous> (node_modules/node-vibrant/src/image/node.ts:9:23)
      at Object.<anonymous> (node_modules/node-vibrant/src/index.ts:2:1)
      at Object.<anonymous> (node_modules/react-palette/dist/react-palette.js:1:161)
      at Object.<anonymous> (src/app/components/PublisherCard/index.tsx:1:1)
      at Object.<anonymous> (src/app/screens/ConfirmPayment/index.tsx:10:1)
      at Object.<anonymous> (src/app/screens/ConfirmPayment/index.test.tsx:4:1)
```

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -7,4 +7,8 @@ module.exports = {
     "^.+.(css|styl|less|sass|scss|png|jpg|ttf|woff|woff2)$":
       "jest-transform-stub",
   },
+  setupFiles: [
+    "jest-webextension-mock"
```
Comment (escapedcat) [inline jest.config.js:11]: Needed to handle [`webextension-polyfill`](https://github.com/getAlby/lightning-browser-extension/blob/master/src/common/lib/msg.ts#L1)

File: jest.config.js [hunk 2]
```diff
@@ -7,4 +7,8 @@ module.exports = {
...
+  ],
+  testEnvironment: "jsdom"
```
Comment (escapedcat) [inline jest.config.js:13]: Handles:
```sh
    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.

    ReferenceError: document is not defined
```

File: package.json [hunk 3]
```diff
@@ -92,12 +92,15 @@
...
+    "@testing-library/jest-dom": "^5.16.2",
+    "@testing-library/react": "^12.1.2",
+    "@testing-library/user-event": "^13.5.0",
     "@types/crypto-js": "^4.1.0",
     "@types/elliptic": "^6.4.14",
+    "@types/jest": "^27.4.0",
     "@types/memoizee": "^0.4.7",
     "@types/pubsub-js": "^1.8.3",
     "@types/react-modal": "^3.13.1",
-    "@types/react-router-dom": "^5.3.3",
```
Comment (escapedcat) [inline package.json:100]: Afaik can be removed because v6 doesn't need this anymore

File: tsconfig.json [hunk 4]
```diff
@@ -68,5 +68,6 @@
     /* Advanced Options */
     "skipLibCheck": true,                           /* Skip type checking of declaration files. */
     "forceConsistentCasingInFileNames": true        /* Disallow inconsistently-cased references to the same file. */
-  }
+  },
+  "include": ["src"]
```
Comment (escapedcat) [inline tsconfig.json:72]: Needed so TS picks up [the dom-types](https://github.com/escapedcat/lightning-browser-extension/blob/chore/584_testing-library-jest-setup/src/setupTests.ts#L5)

File: package.json [hunk 5]
```diff
@@ -135,6 +139,7 @@
     "stream-browserify": "^3.0.0",
     "tailwindcss": "^3.0.15",
     "terser-webpack-plugin": "5.3.0",
+    "ts-node": "^10.4.0",
```
Comment (escapedcat) [inline package.json:142]: To support `jest.config.ts` and `jest.setup.ts`. Got the tip via jest terminal-output. There might be a better way to support this, not sure.

Comment (escapedcat) [inline package.json:142]: To support `jest.config.ts` and `jest.setup.ts`. Got the tip via jest terminal-output. There might be a better way to support this, not sure.

File: src/app/screens/ConfirmPayment/index.test.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+// import { render, screen } from "@testing-library/react";
```
Comment (escapedcat) [inline src/app/screens/ConfirmPayment/index.test.tsx:1]: This keeps failing for reasons mentioned in the PR description.  
Happy to pair with someone on this or remove it for now to get the general setup in first.

File: src/app/screens/ConfirmPayment/index.test.tsx [hunk 7]
```diff
@@ -1,40 +1,42 @@
...
-//       image:
-//         "https://getalby.com/assets/alby-503261fa1b83c396b7ba8d927db7072d15fea5a84d387a654c5d0a2cefd44604.svg",
-//       icon: "https://getalby.com/favicon.ico",
-//     },
-//     external: true,
-//   },
-//   paymentRequest:
-//     "lnbc250n1p3qzycupp58uc2wa29470f98wrxmy4xwuqt8cywjygf5t2cp0s376y7nwdyq3sdqhf35kw6r5de5kueeqg3jk6mccqzpgxqyz5vqsp5wfdmwtv5rmru00ajsnn3f8lzpxa4snug2tmqvc8zj8semr4kjjts9qyyssq83h74pte8nrkqs8sr2hscv5zcdmhwunwnd6xr3mskeayh96pu7ksswa6p7trknlpp6t3js4k6uytxutv5ecgcwaxz7fj4zfy5khjcjcpf66muy",
-// };
+jest.mock("invoices");
```
Comment (escapedcat) [inline src/app/screens/ConfirmPayment/index.test.tsx:6]: Nice! Remind me not to think too complicated :D

File: __mocks__/node-vibrant.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
+const Swatch = (hex) => ({
```
Comment (escapedcat) [inline __mocks__/node-vibrant.ts:1]: Awesome! I'm interested to know how you debugged this.

**Comment by escapedcat** [comment]
It's 🟢  ! Thanks to @dylancom  
@bumi you wanna have a look?

File: src/app/screens/ConfirmPayment/index.tsx [hunk 9]
```diff
@@ -1,25 +1,27 @@
...
 
-type Props = {
+export type Props = {
   origin: OriginData;
   paymentRequest: string;
 };
 
 function ConfirmPayment(props: Props) {
+  console.log("ORIGIN!", props.origin);
+  console.log("paymentRequest!", props.paymentRequest);
```
Comment (bumi) [inline src/app/screens/ConfirmPayment/index.tsx:23]: some console.log leftovers.

**Review by bumi — APPROVED** [review]
I found a console.log leftover but I then it looks good to me! 
super cool to have this setup, thanks for doing it!

**Review by bumi — APPROVED** [review]
@dylancom merge it when you're ready.

**Review by dylancom — APPROVED** [review]

**Pull request merged**